packages/flutter_tools/test/general.shard/test_compiler_test.dart

```diff
@@ -21,85 +24,117 @@ final Platform linuxPlatform = FakePlatform(
 );
 
 void main() {
-  group(TestCompiler, () {
-    Testbed testbed;
-    FakeTestCompiler testCompiler;
-    MockResidentCompiler residentCompiler;
-
-    setUp(() {
-      testbed = Testbed(
-        overrides: <Type, Generator>{
-          Platform: () => linuxPlatform,
-        },
-        setup: () async {
-          globals.fs.file('pubspec.yaml').createSync();
-          globals.fs.file('.packages').writeAsStringSync('flutter_test:flutter_test/');
-          globals.fs.file('test/foo.dart').createSync(recursive: true);
-          residentCompiler = MockResidentCompiler();
-          testCompiler = FakeTestCompiler(
-            BuildMode.debug,
-            false,
-            FlutterProject.current(),
-            residentCompiler,
-          );
-        },
-      );
+  MockResidentCompiler residentCompiler;
+  FileSystem fileSystem;
+
+  setUp(() {
+    fileSystem = MemoryFileSystem.test();
+    fileSystem.file('pubspec.yaml').createSync();
+    fileSystem.file('.packages').writeAsStringSync('flutter_test:flutter_test/');
+    fileSystem.file('test/foo.dart').createSync(recursive: true);
+    fileSystem.file(globalPackagesPath)
```

Comment:
Need to create at the global packages path due to:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/test/test_compiler.dart#L136 . IF this wasn't correct we would throw errors after the test finished.